pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -261,7 +261,9 @@ func (jm *ControllerV2) getJobsToBeReconciled(cronJob *batchv1.CronJob) ([]*batc
 
 	for _, job := range jobList {
 		// If it has a ControllerRef, that's all that matters.
-		if controllerRef := metav1.GetControllerOf(job); controllerRef != nil && controllerRef.Name == cronJob.Name {
+		controllerRef := metav1.GetControllerOf(job)
+		if controllerRef != nil && controllerRef.Name == cronJob.Name &&
+			(controllerRef.Kind == "" || controllerRef.Kind == controllerKind.Kind) {
```

Comment:
why are we interested in empty kinds?

Reply:
Just kept it since I see that we were allowing empty kinds in the past, to avoid a breaking change probably empty kind should be allowed? Though, if the job is run via CRON, it would have the kind as `CronJob` always. Let me know if we can remove the check for Empty kind.